ci-operator/step-registry/ibmcloud/provision/bastionhost/ibmcloud-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,128 @@
+}
+
+function checkCli() {
+  export IBMCLOUD_CLI=ibmcloud
+  export IBMCLOUD_HOME=/output
+  echo "check IBMCLOUD_CLI: ${IBMCLOUD_CLI}..."
+  command -v ${IBMCLOUD_CLI}
+  ${IBMCLOUD_CLI} --version
+  ${IBMCLOUD_CLI} plugin list
+}
+#####################################
+##############Initialize#############
+#####################################
+checkCli
+#create the bastion host based on the info (ibmcloud-provision-vpc ${SHARED_DIR}/customer_vpc_subnets.yaml), output ip info to ${SHARED_DIR}/bastion-info.yaml
+cluster_name="${NAMESPACE}-${JOB_NAME_HASH}"
+bastion_info_yaml="${SHARED_DIR}/bastion-info.yaml"
+
+bastion_name="${cluster_name}-bastion"
+MACHINE_TYPE="bx2-2x8"
+IMAGE="ibm-fedora-coreos-36-stable-2"
+workdir=${ARTIFACT_DIR}
+bastion_ignition_file="${SHARED_DIR}/${cluster_name}-bastion.ign"
+
+if [[ ! -f "${bastion_ignition_file}" ]]; then
+  echo "'${bastion_ignition_file}' not found, ignition-bastionhost step is required, abort." && exit 1
+fi
+
+VPC_CONFIG="${SHARED_DIR}/customer_vpc_subnets.yaml"
+if [[ ! -f "${VPC_CONFIG}" ]]; then
+  echo "Fail to find of VPC info file ${VPC_CONFIG}, abort." && exit 1
+fi
+echo "Reading variables from ${VPC_CONFIG}..."
+vpcName=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.vpcName')
+region="${LEASED_RESOURCE}"
+resource_group=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.resourceGroupName')
+echo "Using region: ${region}  resource_group: ${resource_group} vpc: ${vpcName}"
+
+ibmcloud_login
+
+${IBMCLOUD_CLI} target -g ${resource_group} -r ${region}
+
+subnet=$(${IBMCLOUD_CLI} is vpc ${vpcName} --show-attached --output JSON | jq -c -r '[.subnets[] | select(.name|test("control-plane")) | .name][0]')
+
+if [ "${subnet}"X == X ]; then
+  echo "ERROR" "Fail to get subnet of vpc ${vpcName}, abort" &&  exit 1
+fi
+echo "subnet: ${subnet}"
+zone=$(${IBMCLOUD_CLI} is subnet ${subnet} --output JSON | jq -r '.zone.name')
+echo "zone ${zone}"
+
+sg=$(${IBMCLOUD_CLI} is vpc-sg ${vpcName} --output JSON | jq -r .id)
+echo "sg:" $sg
+
+#####################################
+##########Create Bastion#############
+#####################################
+run_command "${IBMCLOUD_CLI} is security-group-rule-add $sg inbound tcp --remote \"0.0.0.0/0\" --port-min=22 --port-max=22"
+run_command "${IBMCLOUD_CLI} is security-group-rule-add $sg inbound tcp --remote \"0.0.0.0/0\" --port-min=3128 --port-max=3129"
+run_command "${IBMCLOUD_CLI} is security-group-rule-add $sg inbound icmp --remote \"0.0.0.0/0\" --icmp-type 8 --icmp-code 0 "
+run_command "${IBMCLOUD_CLI} is security-group-rule-add $sg inbound tcp --remote \"0.0.0.0/0\" --port-min=5000 --port-max=5000"
+run_command "${IBMCLOUD_CLI} is security-group-rule-add $sg inbound tcp --remote \"0.0.0.0/0\" --port-min=6001 --port-max=6002"
+
+echo "Created bastion instance"
+run_command "${IBMCLOUD_CLI} is instance-create ${bastion_name} ${vpcName} ${zone} ${MACHINE_TYPE} ${subnet} --image ${IMAGE} --user-data "@${bastion_ignition_file}" --output JSON" || exit 1
```

Comment:
Because we enabled `set -o errexit`, I guess we do not need `|| exit 1` any more.